src/webots/nodes/WbRobot.cpp

```diff
@@ -981,6 +983,8 @@ void WbRobot::dispatchAnswer(QDataStream &stream, bool includeDevices) {
       }
     }
   } else {
+    if (mNeedToForceTime)
+      writeAnswer(stream);
```

Comment:
Are you sure this is needed as `writeAnswer()` is called anyhow on line 993?

Reply:
Yes, this is actually the main point of the fix, in the specific reset case we need to make sure that the robot messages are sent before the device ones to update the time.
If we want to avoid calling it a second time, we could add a kind of `if (!mNeedToForceTime)` for the second call but IMO this complexifies the code uselessly (as the second call to writeAnswer will strictly do nothing as all the messages are already sent).

Reply:
OK, and why do we need to send the Robot answer after the device answers in the normal case? I am afraid the reset case will change this and it may have some side effect. If there is no reason for sending the robot answer after, we should send it before in all cases.

Reply:
I did not put it by default to the beginning for this purpose. When doing the reset, since the time is reset to 0 it does not cause any problem. But for the other steps, I am afraid of non-evident side effects of increasing the time before sending the device messages.

Reply:
I see. Thank you for the explanation. I will try to investigate the possible side effects of moving the robot answer before the devices answer, as it seems cleaner (we do that for the configure message).